job/cache.go

```diff
@@ -204,6 +206,12 @@ func (c *LockFreeJobCache) Start(persistWaitTime time.Duration) {
 	// Occasionally, save items in cache to db.
 	go c.PersistEvery(persistWaitTime)
 
+	// Run retention every hour to clean up old job stats entries
+	if jobstatTtl > 0 {
+		c.retentionPeriod = jobstatTtl
+		go c.RetentionEvery(60 * time.Second)
```

Comment:
If we are only going to run this every minute, we might want to change the CLI options to support minutes instead of seconds?